src/coreComponents/physicsSolvers/multiphysics/LagrangianContactSolver.cpp

```diff
@@ -953,11 +999,31 @@ real64 LagrangianContactSolver::CalculateResidualNorm( DomainPartition const & d
 
   MeshLevel const & mesh = *domain.getMeshBody( 0 )->getMeshLevel( 0 );
 
+  NodeManager const & nodeManager = *mesh.getNodeManager();
+
+  arrayView1d< globalIndex const > const & dispDofNumber =
+    nodeManager.getReference< array1d< globalIndex > >( dofManager.getKey( keys::TotalDisplacement ) );
+
   string const & dofKey = dofManager.getKey( viewKeyStruct::tractionString );
   globalIndex const rankOffset = dofManager.rankOffset();
 
-  real64 const momentumResidualNorm = m_solidSolver->CalculateResidualNorm( domain, dofManager, localRhs );
-  real64 const momentumR2 = momentumResidualNorm * momentumResidualNorm;
+  arrayView1d< integer const > const & elemGhostRank = nodeManager.ghostRank();
+
+  RAJA::ReduceSum< parallelDeviceReduce, real64 > localSum0( 0.0 );
+  forAll< parallelDevicePolicy<> >( nodeManager.size(),
+                                    [localRhs, localSum0, dispDofNumber, rankOffset, elemGhostRank] GEOSX_HOST_DEVICE ( localIndex const k )
+  {
+    if( elemGhostRank[k] < 0 )
+    {
+      localIndex const localRow = LvArray::integerConversion< localIndex >( dispDofNumber[k] - rankOffset );
+      for( localIndex dim = 0; dim < 3; ++dim )
+      {
+        localSum0 += localRhs[localRow + dim] * localRhs[localRow + dim];
+      }
+    }
+  } );
+  real64 const momentumR2 = localSum0.get();
```

Comment:
According to
https://github.com/GEOSX/GEOSX/blob/382fd988a2508283c44f2b93107055aa5216285e/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp#L1157
the residual offered by `SolidMechanicsLagrangianFEM.cpp` is a relative one, while I need the absolute value. That's why I recompute it.

The `+0.0` is there just because there are no valid cases in which the initial global residual is zero ... but if you prefer `+1.0`, I can change it!

Reply:
Ok, now you have the `+1.0`.